### DISCUSSION TITLE
How to correctly install imgui on linux/ubuntu "globally".

### DISCUSSION THREAD

**Discussion by waqasulhaqislamabad** [ocornut/imgui#5814] [New Users Build/Link/Run issues ONLY!]
Hey,
I want to install imgui on linux/ubuntu, I want to install it "globally" so that I can include it my projects.

So for example I want to include it like this:
```
#include <imgui/imgui.h>
int main() {
    // Use IMGUI
    return 0;
}
```
I tried adding it using apt-get but "libimgui-dev" was not found, even though it is here:
https://packages.ubuntu.com/impish/libimgui-dev

Thanks.

**Comment by PathogenDavid** [comment]
It is recommended that you don't.

The `libimgui-dev` package is maintained by a third-party and is very out of date. The canonical way to consume Dear ImGui is by copying its files into your project or using a Git Submodule.

That being said, you're meant to consume that package by using `pkg-config`.

**Reply by Micro-fractal** [reply]
If you install Dear ImGui as a git submodule, how do you get the correct header file locations? If I do `#include "imgui/backends/imgui_impl_glfw.h"` after initializing an `imgui` submodule, the compiler complains that it can't find `imgui.h`. I gather that you are supposed to collect the files into your root folder instead of using a git submodule. How does that mesh with the idea of usin a git submodule? I don't know what `pkg-config` is and how it's supposed to work. It would be great if, if this is necessary for using a git submodule, this would be noted in the README in any way.

**Reply by pattakosn** [reply]
> If you install Dear ImGui as a git submodule, how do you get the correct header file locations? If I do `#include "imgui/backends/imgui_impl_glfw.h"` after initializing an `imgui` submodule, the compiler complains that it can't find `imgui.h`. I gather that you are supposed to collect the files into your root folder instead of using a git submodule. How does that mesh with the idea of usin a git submodule? I don't know what `pkg-config` is and how it's supposed to work. It would be great if, if this is necessary for using a git submodule, this would be noted in the README in any way.

I suppose he means to copy paste/git module the required source files and have something like this in your cmake files
```
add_library(imgui STATIC
        # Main Imgui files
        external/imgui/imgui.cpp
        external/imgui/imgui.h
        external/imgui/imgui_draw.cpp
        external/imgui/imgui_tables.cpp
        external/imgui/imgui_widgets.cpp
        external/imgui/imgui_demo.cpp
        # SDL2+OpenGL-specific files
        external/imgui/backends/imgui_impl_sdl.cpp
        external/imgui/backends/imgui_impl_sdl.h
        external/imgui/backends/imgui_impl_opengl3.cpp
        external/imgui/backends/imgui_impl_opengl3.h)
add_library(imgui::imgui ALIAS imgui)
target_link_libraries(imgui
    PUBLIC
    OpenGL::GL
    $<TARGET_NAME_IF_EXISTS:SDL2::SDL2main>
    $<IF:$<TARGET_EXISTS:SDL2::SDL2>,SDL2::SDL2,SDL2::SDL2-static>
)
target_include_directories(imgui PUBLIC external/imgui/ external/imgui/backends/)
```

**Reply by andrijzyn** [reply]
#
For those who have wandered here, this package is currently up to date in the **debian** repos.
[https://packages.debian.org/sid/libimgui-dev](url) @ 

> It is recommended that you don't.
> 
> The `libimgui-dev` package is maintained by a third-party and is very out of date. The canonical way to consume Dear ImGui is by copying its files into your project or using a Git Submodule.
> 
> That being said, you're meant to consume that package by using `pkg-config`.

Edited: The package is over half a year old 😩
